src/renderers/dom/client/wrappers/ReactDOMTextarea.js

```diff
@@ -66,12 +65,14 @@ var ReactDOMTextarea = {
       '`dangerouslySetInnerHTML` does not make sense on <textarea>.'
     );
 
-    // Always set children to the same thing. In IE9, the selection range will
-    // get reset if `textContent` is mutated.
```

Comment:
Not sure, what happens if you have a contenteditable div with text in it?  I wouldn't be surprised if there were another edge condition where this applied for React reconciliation, independent of text areas.

But regardless, it's feels very wrong for getNativeProps to always return the initialValue BECAUSE it knows that the reconciler will use the `textContent` property (as oppose to any other perfectly-valid method) to set the `children` of the textarea.  Correct solution is for the reconciler to properly handle this edge case by virtue of the fact that `setTextContent` properly handles this edge case.